docs/graalvm-compatibility-in-ballerina-libraries.md

```diff
@@ -0,0 +1,185 @@
+
+```
+**********************************************************************************
+* WARNING: Package is not verified with GraalVM.                                 *
+**********************************************************************************
+
+The GraalVM compatibility property has not been defined for the package
+'<package-name>. This could potentially lead to compatibility issues with GraalVM.
+
+To resolve this warning, please ensure that all Java dependencies of this
+package are compatible with GraalVM. Subsequently, update the Ballerina.toml
+file under the section '[platform.<java*>]' with the attribute
+'graalvmCompatible = true'.
+
+**********************************************************************************
+```
+
+### Analyzing the code base for Java dynamic features
+Since the native-image tool assumes the closed-world assumption, the following [dynamic features](https://www.graalvm.org/jdk17/reference-manual/native-image/dynamic-features/) of Java should be handled explicitly through configuration files.
+- [Accessing Resources](https://www.graalvm.org/jdk17/reference-manual/native-image/dynamic-features/Resources/)
+- [Certificate Management](https://www.graalvm.org/jdk17/reference-manual/native-image/dynamic-features/CertificateManagement/)
+- [Dynamic Proxy](https://www.graalvm.org/jdk17/reference-manual/native-image/dynamic-features/DynamicProxy/)
+- [Java Native Interface (JNI)](https://www.graalvm.org/jdk17/reference-manual/native-image/dynamic-features/JNI/)
+- [JCA Security Services](https://www.graalvm.org/jdk17/reference-manual/native-image/dynamic-features/JCASecurityServices/)
+- [Reflection](https://www.graalvm.org/jdk17/reference-manual/native-image/dynamic-features/Reflection/)
+- [URL Protocols](https://www.graalvm.org/jdk17/reference-manual/native-image/dynamic-features/URLProtocols/)
+
+The analysis should be also done on the third-party libraries used in the module. Some third-party libraries might be already GraalVM compatible, this should be verified with the third-party library owners. You can also refer to the configurations for some of the commonly used libraries in [graalvm-reachability-metadata repository](https://github.com/oracle/graalvm-reachability-metadata/tree/master/metadata).
+
+### Test a sample application with GraalVM
+#### Prerequisites
+1. Latest [Ballerina Swan Lake](https://ballerina.io/downloads/) distribution
+   >**Note:** If you are using macOS with an ARM64 processor, then, install Ballerina using the [ARM64 installer](https://ballerina.io/downloads/).
+
+2. A text editor
+   >**Tip:** Preferably, <a href="https://code.visualstudio.com/" target="_blank">Visual Studio Code</a> with the  <a href="https://wso2.com/ballerina/vscode/docs/" target="_blank">Ballerina extension</a> installed.
+
+3. GraalVM installed and configured appropriately
+   
+#### Configure GraalVM locally
+1. Install GraalVM using [SDKMAN!](https://sdkman.io/). 
```

Comment:
Is this a required step? I think this should be a tip rather than a step. The step should be to download/configure the GraalVM.

Reply:
Ack

Reply:
Fixed via [39340f8](https://github.com/ballerina-platform/ballerina-standard-library/pull/4792/commits/39340f893109eb1e825829eca1ab29425cc18f30)